File: test_import/example_project_cc_via_bazel/src/slog_test.cc
// Copyright 2022 Woven Planet Holdings
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include <gtest/gtest.h>
#include <slog_cc/analysis_tools/tracing/slog_trace_subscriber.h>
#include <slog_cc/buffer/buffer.h>
#include <slog_cc/slog.h>

TEST(SlogTest, basic) {
  SLOG(INFO) << "foo";
  slog::SlogContext::getInstance()->waitAsyncSubscribers();
  {
    slog::SlogBuffer slog_buffer(slog::SlogContext::getInstance());
    {
      auto buffer_data = slog_buffer.flush();
      EXPECT_EQ(0, buffer_data.records.size());
      EXPECT_EQ(2, buffer_data.call_sites.size());
    }

    SLOG(INFO) << "bar";
    slog::SlogContext::getInstance()->waitAsyncSubscribers();
    {
      auto buffer_data = slog_buffer.flush();
      EXPECT_EQ(1, buffer_data.records.size());
      EXPECT_EQ(3, buffer_data.call_sites.size());
    }
    {
      auto buffer_data = slog_buffer.flush();
      EXPECT_EQ(0, buffer_data.records.size());
      EXPECT_EQ(3, buffer_data.call_sites.size());
    }
  }
  SLOG(INFO) << "no segfaults";
}

TEST(SlogTest, chrome_tracing_build) {
  // While slog trace subscriber object exists it will append chrome tracing
  // records to the provided json file. On destruction it will finalize the
  // file.
  auto slog_trace_subscriber =
      slog::CreateSlogTraceSubscriber("/tmp/slog-trace.json");

  // Actual code with slogs body.
  {
    SLOG_SCOPE("foo_scope");
    { SLOG_SCOPE("bar_scope"); }
  }

  // Need to wait all slog records to be processed to make sure all records were
  // flushed and written to the slog json file.
  slog::SlogContext::getInstance()->waitAsyncSubscribers();
}

File: test_import/example_project_py_via_bazel/src/slog_test.py
# Copyright 2022 Woven Planet Holdings
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import json
import unittest

from slog_py import slog
from slog_py.slog_pybind import SlogBuffer, SlogCallSite, SlogRecord


class SlogPyTest(unittest.TestCase):
    def test_severities(self):
        with SlogBuffer(slog.SlogContext.get_instance()) as slog_buffer:
            slog.info('Hello World')
            slog_buffer.wait_slog_queue()
            records = slog_buffer.flush().records

            self.assertEqual(1, len(records))
            self.assertEqual(2, int(json.loads(str(records[0]))['severity']))
            self.assertEqual('Hello World', json.loads(
                str(records[0]))['tags'][0]['valueString'])


if __name__ == '__main__':
    unittest.main()
